packages/datadog_flutter_plugin/test/rum/rum_user_action_detector_test.dart

```diff
@@ -514,4 +537,63 @@ void main() {
     verify(() => mockRum.addAction(RumActionType.tap, 'Tab(Rainy Days)'));
     verifyNoMoreInteractions(mockRum);
   });
+
+  testWidgets(
+      'tap custom widget reports tap to RUM with unknown annotation by default',
+      (tester) async {
+    final mockRum = MockDdRum();
+
+    await tester.pumpWidget(_buildSimpleApp(
+      mockRum,
+      _TestWidget(),
+    ));
+
+    final button = find.byType(_TestWidget);
+    await tester.tap(button);
+
+    verify(() => mockRum.addAction(RumActionType.tap, 'TestWidget(unknown)'));
+  });
+
+  testWidgets('tap custom widget reports tap to RUM and search for better text',
```

Comment:
I think we should also have tests for the `searchForBetter` and `searchForText` options work as expected.